developer/src/kmcmplib/tests/gtest-compiler-test.cpp

```diff
@@ -556,7 +560,106 @@ TEST_F(CompilerTest, GetDelimitedString_test) {
 }
 
 // LinePrefixType GetLinePrefixType(PKMX_WCHAR *p)
-// int LineTokenType(PKMX_WCHAR *str)
+
+TEST_F(CompilerTest, LineTokenType_test) {
+    KMX_WCHAR str[LINESIZE];
+    PKMX_WCHAR p = nullptr;
+
+    // T_BLANK, lptOther, empty string
+    u16cpy(str, u"");
+    p = str;
+    EXPECT_EQ(T_BLANK, LineTokenType(&p));
+
+    // T_BLANK, lptOther, one space
+    u16cpy(str, u" ");
+    p = str;
+    EXPECT_EQ(T_BLANK, LineTokenType(&p));
+
+    // T_BLANK, mismatched prefix, CKF_KEYMAN, lptKeymanWebOnly
+    u16cpy(str, u"$keymanweb:");
+    p = str;
+    kmcmp::CompileTarget = CKF_KEYMAN;
+    EXPECT_EQ(T_BLANK, LineTokenType(&p));
+
+    // T_BLANK, mismatched prefix, CKF_KEYMANWEB, lptKeymanOnly
+    u16cpy(str, u"$keymanonly:");
+    p = str;
+    kmcmp::CompileTarget = CKF_KEYMANWEB;
+    EXPECT_EQ(T_BLANK, LineTokenType(&p));
+
+    // T_BLANK, nothing after prefix
+    u16cpy(str, u"$keyman:");
+    p = str;
+    kmcmp::CompileTarget = CKF_KEYMAN;
+    EXPECT_EQ(T_BLANK, LineTokenType(&p));
+
+    // T_STORE (=T_W_START)
+    u16cpy(str, u"store(b)");
+    p = str;
+    EXPECT_EQ(T_STORE, LineTokenType(&p));
+    EXPECT_EQ(u16len(u"store"), p - str);
+    EXPECT_TRUE(!u16cmp(p, u"(b)"));
+
+    // T_BITMAPS (=T_W_END)
+    u16cpy(str, u"bitmaps \"b\"");
+    p = str;
+    EXPECT_EQ(T_BITMAPS, LineTokenType(&p));
+    EXPECT_EQ(u16len(u"bitmaps "), p - str);
+    EXPECT_TRUE(!u16cmp(p, u"\"b\""));
+
+    // T_STORE, preceeded by one space
+    u16cpy(str, u" store(b)");
+    p = str;
+    EXPECT_EQ(T_STORE, LineTokenType(&p));
+    EXPECT_EQ(u16len(u" store"), p - str);
+    EXPECT_TRUE(!u16cmp(p, u"(b)"));
+
+    // T_STORE, preceeded by two spaces
+    u16cpy(str, u"  store(b)");
+    p = str;
+    EXPECT_EQ(T_STORE, LineTokenType(&p));
+    EXPECT_EQ(u16len(u"  store"), p - str);
+    EXPECT_TRUE(!u16cmp(p, u"(b)"));
+
+    // T_STORE, followed by one space
+    u16cpy(str, u"store (b)");
+    p = str;
+    EXPECT_EQ(T_STORE, LineTokenType(&p));
+    EXPECT_EQ(u16len(u"store "), p - str);
+    EXPECT_TRUE(!u16cmp(p, u"(b)"));
+
+    // T_STORE, followed by two spaces
+    u16cpy(str, u"store  (b)");
+    p = str;
+    EXPECT_EQ(T_STORE, LineTokenType(&p));
+    EXPECT_EQ(u16len(u"store  "), p - str);
+    EXPECT_TRUE(!u16cmp(p, u"(b)"));
+
+    // T_COMMENT
+    u16cpy(str, u"c ");
+    p = str;
+    EXPECT_EQ(T_COMMENT, LineTokenType(&p));
+    EXPECT_EQ(0, p - str);
+
+    // comment without following space
+    u16cpy(str, u"c");
+    p = str;
+    EXPECT_EQ(T_UNKNOWN, LineTokenType(&p));
+    EXPECT_EQ(0, p - str);
```

Comment:
Hmm, I would have thought that this should be treated as a comment. It turns out that `ReadLine()` strips the comment before it ever gets to `LineTokenType()`, so we don't get failure:

https://github.com/keymanapp/keyman/blob/bcec4461448f78498cac2b3711081546705b200a/developer/src/kmcmplib/src/Compiler.cpp#L3404-L3410

(In `ReadLine()` the input has a trailing `\n` so `iswspace(*(p+1))` is true.)

Reply:
I have left as is, except to add a comment to the test cf. `ReadLine()`.  Should I open an issue?

Reply:
Yeah, probably open an issue, add the low priority label to it.